app/code/Magento/Sales/Model/EmailSenderHandler.php

```diff
@@ -66,8 +66,10 @@ public function __construct(
     public function sendEmails()
     {
         if ($this->globalConfig->getValue('sales_email/general/async_sending')) {
+            $this->entityCollection->clear();
```

Comment:
As we discussed - it's not good idea to clean collection every time. It's better to fix root of the issue - this object should NOT be shared. Please add `shared="false"` to https://github.com/magento/magento2/blob/2.2-develop/app/code/Magento/Sales/etc/di.xml#L355-L395